x-pack/plugins/cases/public/components/files/file_type.test.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { JsonValue } from '@kbn/utility-types';
+
+import { screen } from '@testing-library/react';
+
+import type { ExternalReferenceAttachmentViewProps } from '../../client/attachment_framework/types';
+import type { AppMockRenderer } from '../../common/mock';
+
+import { AttachmentActionType } from '../../client/attachment_framework/types';
+import { FILE_ATTACHMENT_TYPE } from '../../../common/api';
+import { createAppMockRenderer } from '../../common/mock';
+import { basicCase, basicFileMock } from '../../containers/mock';
+import { getFileType } from './file_type';
+import userEvent from '@testing-library/user-event';
+
+describe('getFileType', () => {
+  const fileType = getFileType();
+
+  it('invalid props return blank FileAttachmentViewObject', () => {
+    expect(fileType).toStrictEqual({
+      id: FILE_ATTACHMENT_TYPE,
+      icon: 'document',
+      displayName: 'File Attachment Type',
+      getAttachmentViewObject: expect.any(Function),
+      hideDefaultActions: true,
+    });
+  });
+
+  describe('getFileAttachmentViewObject', () => {
+    let appMockRender: AppMockRenderer;
+
+    const attachmentViewProps = {
+      externalReferenceId: basicFileMock.id,
+      externalReferenceMetadata: { files: [basicFileMock] },
+      caseData: { title: basicCase.title, id: basicCase.id },
+    } as unknown as ExternalReferenceAttachmentViewProps;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('event renders a clickable name if the file is an image', async () => {
+      appMockRender = createAppMockRenderer();
+
+      // @ts-ignore
```

Comment:
Are these `ts-ignore` because the `event` can be a string or component?

Reply:
Just because `event` is possibly `undefined`.